### PR TITLE
FalkorDB Multi Graph ability

### PR DESCRIPTION
**Pull request by galshubeli**
This PR introduces the ability to use multi-graph with the FalkorDB graph database.

### CODE REVIEW

File: src/memary/agent/base_agent.py [hunk 1]
```diff
@@ -51,19 +51,20 @@ class Agent(object):
...
     def __init__(
         self,
-        name,
+        agent_name,
         memory_stream_json,
         entity_knowledge_store_json,
         system_persona_txt,
         user_persona_txt,
         past_chat_json,
+        user_id='__falkor__',
```
Comment (swilly22) [inline src/memary/agent/base_agent.py:60]: is that an appropriate user id ?

File: src/memary/agent/base_agent.py [hunk 2]
```diff
@@ -83,7 +84,7 @@ def __init__(
         if self.falkordb_url is not None:
             from llama_index.graph_stores.falkordb import FalkorDBGraphStore
             # initialize FalkorDB graph resources
-            self.graph_store = FalkorDBGraphStore(self.falkordb_url, decode_responses=True)
+            self.graph_store = FalkorDBGraphStore(self.falkordb_url, database=user_id, decode_responses=True)
```
Comment (swilly22) [inline src/memary/agent/base_agent.py:87]: database is basically the graph key ?

File: streamlit_app/app.py [hunk 3]
```diff
@@ -117,7 +107,18 @@ def get_models(llm_models, vision_models):
...
+    user_id = st.text_input(
+        "Enter a user ID (Available only with FalkorDB)",
+    )
+with col2:
+    st.write("")
+    submit = st.button("Switch User ID")
 
+if not submit:
+    user_id = "__falkor__"
+
```
Comment (swilly22) [inline streamlit_app/app.py:121]: Add comment explaining this logic

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/memary/agent/base_agent.py (hunk 1 above)
Comment (galshubeli) [reply src/memary/agent/base_agent.py:60]: changed to 'falkor'

File: src/memary/agent/base_agent.py (hunk 2 above)
Comment (galshubeli) [reply src/memary/agent/base_agent.py:87]: yes

**Review by swilly22 — APPROVED** [review]

File: streamlit_app/app.py [hunk 4]
```diff
@@ -117,7 +107,23 @@ def get_models(llm_models, vision_models):
...
 
+# Create a text input field for user ID
+with col1:
+    user_id = st.text_input(
+        "Enter a user ID (Available only with FalkorDB)",
+    )
+# Create a button to apply the switch user ID
+with col2:
+    st.write("")
+    submit = st.button("Switch User ID")
```
Comment (kingjulio8238) [inline streamlit_app/app.py:121]: "Switch Agent ID" would be more suitable.

File: README.md [hunk 5]
```diff
@@ -115,6 +115,33 @@ chat_agent = ChatAgent(
 ```
 Pass in subset of `['search', 'vision', 'locate', 'stocks']` as `include_from_defaults` for different set of default tools upon initialization.
 
+### Multi-Graph
```
[the remark this answers is not in the document]
Comment (galshubeli) [reply README.md:120]: Please let me know what you think about the step I added.

**Review by kevinl424 — APPROVED** [review]
LGTM!

**Review by kingjulio8238 — APPROVED** [review]
LGTM!

**Review by kingjulio8238 — APPROVED** [review]
All comments resolved

**Review by Muaytie666 — APPROVED** [review]

**Pull request merged**